File: LAB 2/Python/BRESEN_CIRCLE.py
import matplotlib.pyplot as plt
import numpy as np

RADIUS = 8
CX, CY = 10, 10

plt.figure()
plt.grid(True)
plt.axis('equal')
plt.title('Basic Circle Scan Conversion')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

THETA = np.linspace(0, 2*np.pi, 100)
X_CIRCLE = CX + RADIUS * np.cos(THETA)
Y_CIRCLE = CY + RADIUS * np.sin(THETA)
plt.plot(X_CIRCLE, Y_CIRCLE, 'b-')

X_VALS, Y_VALS = [], []

for X in range(CX - RADIUS, CX + RADIUS + 1):
    Y = round(CY + np.sqrt(RADIUS**2 - (X - CX)**2))
    X_VALS.extend([X, X])
    Y_VALS.extend([Y, 2*CY - Y])

plt.plot(X_VALS, Y_VALS, 'ks', markersize=8, markerfacecolor='k')
plt.plot(CX, CY, 'ro', markersize=10, markerfacecolor='r')
plt.show()

File: LAB 2/MATLAB/BRESEN_CIRCLE.m
clc; clear; close all;

RADIUS = 8;
CENTER_X = 10;
CENTER_Y = 10;

figure;
hold on;
grid on;
axis equal;
title('Bresenham''s Circle Algorithm');
xlabel('X-axis'); ylabel('Y-axis');

X = 0;
Y = RADIUS;
D = 3 - 2 * RADIUS;

while X <= Y
    plot(CENTER_X + X, CENTER_Y + Y, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X - X, CENTER_Y + Y, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X + X, CENTER_Y - Y, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X - X, CENTER_Y - Y, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X + Y, CENTER_Y + X, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X - Y, CENTER_Y + X, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X + Y, CENTER_Y - X, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    plot(CENTER_X - Y, CENTER_Y - X, 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');

    if D < 0
        D = D + 4 * X + 6;
    else
        D = D + 4 * (X - Y) + 10;
        Y = Y - 1;
    end
    X = X + 1;
end

THETA = 0:0.01:2*pi;
X_CIRCLE = CENTER_X + RADIUS * cos(THETA);
Y_CIRCLE = CENTER_Y + RADIUS * sin(THETA);
plot(X_CIRCLE, Y_CIRCLE, 'b-');
